flake.nix

```diff
@@ -7,43 +7,72 @@
   };
 
   outputs = { self, nixpkgs, utils, rust }:
+    let
+      pname = "mdzk";
+      version =
+        (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package.version;
+    in
+    {
+      overlays.default = nixpkgs.lib.composeManyExtensions [
+        rust.overlay
+        (final: prev:
+          let
+            rust-toolchain = final.rust-bin.selectLatestNightlyWith (toolchain:
+              toolchain.default.override {
+                extensions = [ "rust-std" "rust-src" ];
+              }
+            );
+          in
+          {
+            rustc = rust-toolchain;
+            cargo = rust-toolchain;
+            mdzk = import ./nix/package.nix {
+              inherit pname version;
+              pkgs = final;
+            };
+          })
+      ];
+    } //
     utils.lib.eachDefaultSystem (system:
       let
-        pname = "mdzk";
-        version =
-          (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package.version;
-
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ (import rust) ];
+          overlays = [
+            self.overlays.default
+          ];
         };
 
-        rust-toolchain = pkgs.rust-bin.selectLatestNightlyWith (toolchain:
-          toolchain.default.override {
-            extensions = [ "rust-std" "rust-src" ];
-          });
-
-        mdzk-pkg = pkgs.callPackage ./default.nix {
-          inherit pkgs pname version rust-toolchain;
-        };
-      in rec {
+        inherit (pkgs) mdzk;
+      in
+      rec {
         # `nix build`
-        packages.${pname} = mdzk-pkg;
+        packages.${pname} = mdzk;
         packages.default = packages.${pname};
 
         # `nix run`
         apps.${pname} = utils.lib.mkApp { drv = packages.${pname}; };
-        defaultApp = apps.${pname};
+        apps.default = apps.${pname};
 
         # `nix develop`
         devShells.default =
-          pkgs.mkShell { nativeBuildInputs = with pkgs; [ rust-toolchain ]; };
+          pkgs.mkShell {
+            nativeBuildInputs = with pkgs; [
+              # rust
+              cargo
+              clippy
+              rust-analyzer
+              rustc
+
+              # misc
```

Comment:
Let's just add `fswatch` (needed for live-updating docs) here as well, so that we don't need two dev-shells.